### PR TITLE
Make soldierOn loop sleep configurable

### PR DESCRIPTION
**Pull request by nkramer44**
In `SimpleStreamSender`, we currently sleep for 100 ms in between packet send tries while ramping up.  This should be configurable per #406 

To fix this, I added an `Optional<UnsignedLong>` field to `SimpleStreamSender` and `SendMoneyAggregator` to configure this value, which gets defaulted to 100ms.  Any STREAM sender can configure this value via an env property to instantiate a `SimpleStreamSender`

### CODE REVIEW

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java [hunk 1]
```diff
@@ -120,33 +146,52 @@ public SimpleStreamSender(final StreamEncryptionService streamEncryptionService,
...
+   *                                end-to-end STREAM packet data (i.e., packets that should only be visible between
+   *                                sender and receiver).
+   * @param link                    A {@link Link} that is used to send ILPv4 packets to an immediate peer.
+   * @param executorService         executorService to run the payments
+   * @param sendPacketSleep amount of time for a thread to sleep before sending more packets
+   */
+  public SimpleStreamSender(
+    final StreamEncryptionService streamEncryptionService,
+    final Link link,
+    ExecutorService executorService,
```
Comment (sappenin) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: Minor nit:
```suggestion
final ExecutorService executorService,
```

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java [hunk 2]
```diff
@@ -120,33 +146,52 @@ public SimpleStreamSender(final StreamEncryptionService streamEncryptionService,
...
   /**
    * Required-args Constructor.
    *
-   * @param streamEncryptionService A {@link StreamEncryptionService} used to encrypt and decrypted end-to-end STREAM
+   * @param streamEncryptionService An instance of {@link StreamEncryptionService} used to encrypt and decrypted
+   *                                end-to-end STREAM packet data (i.e., packets that should only be visible between
+   *                                sender and receiver).
+   * @param link                    A {@link Link} that is used to send ILPv4 packets to an immediate peer.
+   * @param executorService         executorService to run the payments
+   * @param sendPacketSleep amount of time for a thread to sleep before sending more packets
```
Comment (sappenin) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: Minor nit: formatting

```suggestion
   * @param         sendPacketSleep amount of time for a thread to sleep before sending more packets
```

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java [hunk 3]
```diff
@@ -120,33 +146,52 @@ public SimpleStreamSender(final StreamEncryptionService streamEncryptionService,
...
+  /**
+   * Required-args Constructor.
+   *  @param streamEncryptionService A {@link StreamEncryptionService} used to encrypt and decrypted end-to-end STREAM
    *                                packet data (i.e., packets that should only be visible between sender and
    *                                receiver).
    * @param link                    A {@link Link} that is used to send ILPv4 packets to an immediate peer.
    * @param executorService         A {@link ExecutorService} to run the payments.
    * @param streamConnectionManager A {@link StreamConnectionManager} that manages connections for all senders and
-   *                                receivers in this JVM.
+   * @param sendPacketSleep amount of time for a thread to sleep before sending more packets
```
Comment (sappenin) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]:
```suggestion
   * @param                         sendPacketSleep amount of time for a thread to sleep before sending more packets
```

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java [hunk 4]
```diff
@@ -120,33 +146,52 @@ public SimpleStreamSender(final StreamEncryptionService streamEncryptionService,
...
-      final StreamEncryptionService streamEncryptionService,
-      final Link link,
-      final ExecutorService executorService,
-      final StreamConnectionManager streamConnectionManager
-  ) {
+    final StreamEncryptionService streamEncryptionService,
+    final Link link,
+    final ExecutorService executorService,
+    final StreamConnectionManager streamConnectionManager,
+    Optional<UnsignedLong> sendPacketSleep) {
```
Comment (sappenin) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]:
```suggestion
  final Optional<UnsignedLong> sendPacketSleep
  ) {
```

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java [hunk 5]
```diff
@@ -298,6 +347,9 @@ default int totalPackets() {
       this.receiverDenomination = Optional.empty();
 
       this.unrecoverableErrorEncountered = new AtomicBoolean(false);
+
+      Objects.requireNonNull(sendPacketSleep);
+      this.sendPacketSleep = sendPacketSleep.orElse(UnsignedLong.valueOf(100L)).longValue();
```
Comment (sappenin) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: There was talk that we should make this default much lower, like 10 -- @nhartner @theotherian does that sound OK?

File: stream-parent/stream-client/src/test/java/org/interledger/stream/sender/SimpleStreamSenderTests.java [hunk 6]
```diff
@@ -65,7 +66,13 @@ public void constructThreeArgWithNullLink() {
   @Test
   public void constructThreeArgWithNullExecutor() {
     expectedException.expect(NullPointerException.class);
-    new SimpleStreamSender(streamEncryptionServiceMock, linkMock, null);
+    new SimpleStreamSender(streamEncryptionServiceMock, linkMock, (ExecutorService) null);
+  }
+
+  @Test
+  public void constructThreeArgWithNullSleep() {
```
Comment (sappenin) [inline stream-parent/stream-client/src/test/java/org/interledger/stream/sender/SimpleStreamSenderTests.java]: Not sure if it's worth it, but should we make a unit test with a simulated Clock to test the loop?

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java (hunk 5 above)
Comment (nkramer44) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: I set the default to 10ms.  After running some tests, there doesn't seem to be a difference in total send time between `SimpleStreamSender`s with a 10ms sleep and 5ms sleep.

We should maybe do some more testing to figure out the most optimal sleep time, but that's probably overkill for now

File: stream-parent/stream-client/src/test/java/org/interledger/stream/sender/SimpleStreamSenderTests.java (hunk 6 above)
Comment (nkramer44) [reply stream-parent/stream-client/src/test/java/org/interledger/stream/sender/SimpleStreamSenderTests.java]: Check out the latest commit to `SimpleStreamSenderIT`.  I send the same payment through a `SimpleStreamSender` with sleep time set to 100ms and a `SimpleStreamSender` with sleep time set to 5ms.  Total time to send the payments (in this case 7 packets) was much much lower with a smaller sleep time

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java (hunk 5 above)
Comment (sappenin) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: Per the Quilt community call - let's explore 20ms or 50ms and just double-check that 10ms seems to be the best sleep time.

File: stream-parent/stream-client/src/test/java/org/interledger/stream/sender/SimpleStreamSenderTests.java (hunk 6 above)
Comment (sappenin) [reply stream-parent/stream-client/src/test/java/org/interledger/stream/sender/SimpleStreamSenderTests.java]: Per the ILP community call, consensus is that this is probably not worth the effort.

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java (hunk 5 above)
Comment (nkramer44) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: Ran a bunch of different sleep times.  10ms always seems to execute faster than anything > 10ms.  Once you get below 10ms, it's a wash which one will be faster.

Note this was tested on a payment that sends roughly 20 packets.

**Pull request merged**